qiskit_experiments/curve_analysis/drawer_mixin.py

```diff
@@ -0,0 +1,512 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Curve drawing functionality for analysis."""
+
+import functools
+from typing import Callable, Set, List, Dict, Sequence, Union, Optional
+
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.ticker import FuncFormatter, ScalarFormatter
+from qiskit.utils import detach_prefix
+from uncertainties import unumpy as unp, UFloat
+
+from qiskit_experiments.curve_analysis.curve_data import FitData
+from qiskit_experiments.framework import AnalysisResultData, Options
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+
+
+class CurveDrawerMixin:
+    """Mixin to provide drawing functionalities for curve fit result."""
+
+    # Internal state for drawing options
+    _draw_options: Options
+    _set_draw_options: Set
+
+    def __init_subclass__(cls, **kwargs):
+        """Initialize drawing options."""
+        super().__init_subclass__(**kwargs)
+
+        # This is a hack to let the mixin have internal state for drawing option.
+        @functools.wraps(cls.__init__, assigned=("__annotations__",))
+        def __new__(cls, *args, **kwargs):
+            instance = super().__new__(cls, *args, **kwargs)
+            setattr(instance, "_default_draw_options", cls._default_draw_options)
+            setattr(instance, "_set_draw_options", set())
+            return instance
+
+        cls.__new__ = __new__
+
+    def _initialize_canvas(self) -> Axes:
+        """Initialize matplotlib axis.
+
+        Curve analysis drawer supports 2D sub-graph visualization.
+        This first checks the drawing options if matplotlib axis is provided by users.
+        Axis is initialized only when axis is not provided.
+        Initialized axis is set to the drawing options.
+
+        Returns:
+            Handler to matplotlib axis object. This may be initialized with inset axes.
+        """
+        # Create axis if empty
+        if not self.draw_options.axis:
+            axis = get_non_gui_ax()
+
+            # Set figure size if axis is not provided.
+            # Otherwise user may set the figure size by oneself.
+            # Then avoid applying draw defaults to avoid override user's preference.
```

Comment:
This is a bit confusing. Do we need this comment? The code looks clean enough.

Reply:
I moved the comment to option description. Indeed this should not be a comment for developers.
https://github.com/Qiskit/qiskit-experiments/pull/738/commits/434b8949d51ee84d4021f56a4783e9a58ca938f6